api/add.go

```diff
@@ -0,0 +1,115 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/weaveworks/pctl/pkg/bootstrap"
+	"github.com/weaveworks/pctl/pkg/catalog"
+	"github.com/weaveworks/pctl/pkg/git"
+	"github.com/weaveworks/pctl/pkg/install"
+	"github.com/weaveworks/pctl/pkg/log"
+)
+
+// AddProfileOpts defines settings for adding profiles.
+type AddProfileOpts struct {
+	URL           string
+	Branch        string
+	SubName       string
+	Namespace     string
+	ConfigMap     string
+	Dir           string
+	Path          string
+	GitRepository string
+	ProfilePath   string
+	CatalogClient catalog.CatalogClient
+	GitClient     git.Git
+}
```

Comment:
not all of these options can be configured at the same time right? E.g. ProfilePath vs URL. Is there a way we could convey this through the API?

Reply:
Yeah, I see in value in that.

Reply:
I suspect that `Dir` isn't part of this too?

I don't think all these fields are really part of the `AddProfileOpts` ?

Reply:
The reason they are here, is that without it, there is little value in having this function. I tried to put as much as possible from the CLI into the API. Otherwise, just construct the manager.Install yourself, which wego is already doing. :)

I get how this seems as not really part of an API. But without these things, I don't think there is too much value in extracting AddProfile.

Reply:
I do think there's value, for folks _outside_ of pctl, consider this as an implementation of the Facade pattern?

Reply:
> I suspect that `Dir` isn't part of this too?
> 
> I don't think all these fields are really part of the `AddProfileOpts` ?

do you mean that some are mandatory vs optional?

Reply:
Basically, yes...

In my view, all the exposed functions for the profiles client should take the directory to operate from as the _first_ parameter, there may be other parameters that should be in this list too

Reply:
Alright, will do it like that. :)

Reply:
We are going to duplicate some checks in the api and the cli then I guess.